### ISSUE TITLE
Helper `format-date` - Why are options `dateStyle` and `timeStyle` not supported?

### ISSUE THREAD

**Issue by MrChocolatine** [ember-intl/ember-intl#1564]
- [X] I am on the [latest](https://github.com/ember-intl/ember-intl/releases/latest) ember-intl version
- [X] I have searched the [issues](https://github.com/ember-intl/ember-intl/issues) of this repo and believe that this is not a duplicate

### Environment

- Ember Version: 3.24.0
- Ember CLI Version: 3.24.0
- Ember Intl Version: 5.6.2
- Browser(s): Chromium, Chrome or Safari
- Node Version: 12.22.1 and 14.16.1

### Steps to Reproduce

- Clone this demo for the demonstration
  https://github.com/MrChocolatine/ember-intl-issue-helper-format-date

- Enter the folder and run the project

```
ember serve
```

- Open your browser at http://localhost:4200

- Notice the difference

```
ember-intl
13/05/2021

Manually
Thursday, 13 May 2021 at 18:47:51 BST
```

The interesting code is in this commit:  
https://github.com/MrChocolatine/ember-intl-issue-helper-format-date/commit/5a6ae8b276f3f0edb6aac86cbeba8ce587e5711c

`ember-intl` relies on `Intl` so why any other option is supported but not `dateStyle` and `timeStyle`?

**Comment by MrChocolatine** [comment]
Bump

**Comment by MrChocolatine** [comment]
Nobody?

**Comment by longlho** [comment]
sry we'll try to take a look as soon as we can. At the same time PR's welcome as well :)

**Comment by buschtoens** [comment]
Interestingly, this used to work in 504f09089ae9481d2182e792fcbc4dc57edf6e0f, which is still deployed as [`master`](https://ember-intl.github.io/ember-intl/versions/master/) to the docs. It's _not_ working in [`v5.7.0`](https://ember-intl.github.io/ember-intl/versions/v5.7.0/) (ca5ec90c927e20f7aadfe4031ba28de3e31dafe2), the first release after the renaming from `master` to `main`, but also not in [`v5.6.2`](https://ember-intl.github.io/ember-intl/versions/v5.6.2/) (8b7f7ac20e6e3205deecf1a54ffa05519572dd49), the last release before `master`.

I ran this snippet in the console:

```js
const { owner } = Ember.Namespace.NAMESPACES_BY_ID.dummy.__container__;
const intl = owner.lookup('service:intl');
intl.formatDate(new Date(), {
  dateStyle: 'full',
  timeStyle: 'long',
});
```

This prints:

- `v5.7.0`: `"7/7/2021"`
- `master`: `"Wednesday, July 7, 2021 at 6:40:34 PM GMT+2"`
- `v5.6.2`: `"7/7/2021"`

So something in the range https://github.com/ember-intl/ember-intl/compare/504f09089ae9481d2182e792fcbc4dc57edf6e0f...v5.6.2 fixed it, where as something in https://github.com/ember-intl/ember-intl/compare/v5.7.0...504f09089ae9481d2182e792fcbc4dc57edf6e0f broke it again.

**Comment by buschtoens** [comment]
Actually, I think the docs have a major caching issue. I'm seeing `ember-intl/-private/formatters/-format-datetime` in the [`vendor.js`](https://ember-intl.github.io/ember-intl/versions/v5.7.0/assets/vendor.js) of `v5.7.0`, even though that file has been removed in 6f1a7cd6c9eba23efb4a5f1209e457c8177a4952.

**Comment by buschtoens** [comment]
Running the code snippet against `v5.7.0` ca5ec90c927e20f7aadfe4031ba28de3e31dafe2 built locally works as expected:


```js
const { owner } = Ember.Namespace.NAMESPACES_BY_ID.dummy.__container__;
const intl = owner.lookup('service:intl');
intl.formatDate(new Date(), {
  dateStyle: 'full',
  timeStyle: 'long',
});
```

This prints: `"Wednesday, July 7, 2021 at 7:07:53 PM GMT+2"`

@MrChocolatine Can you please update to `v5.7.0` and confirm, that it's working for you as well?

**Comment by MrChocolatine** [comment]
Thanks for the feedback.

@buschtoens 
On my test repository I upgraded `ember-intl` to `^5.7.0` and tried a fixed version `5.7.0` and I still have the same issue: (follow the same steps as the initial post)

https://github.com/MrChocolatine/ember-intl-issue-helper-format-date/commits/master

**Comment by buschtoens** [comment]
I think I found the issue. The release `v5.7.0` was built incorrectly.

`v5.7.0` somehow reintroduced `-format-datetime.ts` in https://github.com/ember-intl/ember-intl/tree/v5.7.0/addon/-private/formatters. It is also present in the [`ember-intl@5.7.0` npm package](https://unpkg.com/browse/ember-intl@5.7.0/addon/-private/formatters/-format-datetime.ts).

The next commit 791a8c97fb9442cfe692fd80470d82ed2a1b646b `Merge tag 'v5.7.0'` (current `main`) then removed the file again: https://github.com/ember-intl/ember-intl/tree/791a8c97fb9442cfe692fd80470d82ed2a1b646b/addon/-private/formatters

I'll publish a `5.7.1`, which should resolve this.

**Comment by buschtoens** [comment]
Unfortunately, the `main` branch is in a broken state. The version got incremented to `6.0.0` and published as a `beta` after #1530 was merged. This PR introduced a `@dependentKeyCompat` decorator on `_locale` and AFACIT, this is one of the causes for the broken `main` branch. Nevertheless, I've published a `6.0.0-beta.1` with the latest (broken) state of `main`.

I would rather invest time into stabilizing `6.0.0`, as it fixes this issue as a side-effect, than fixing it for `5.7.0` and pushing out an intermediary patch release.
